src/book/entities/splitOps.ts

```diff
@@ -0,0 +1,7 @@
+import Split from './Split';
+
+export const splitOps = {
+  isSameCommodity: (a: Split, b: Split) => (
+    a.account?.commodity?.guid === b.account?.commodity?.guid
+  ),
+};
```

Comment:
maybe this could be part of the Commodity entity compare method and then have another in the Split as `sameCommodity`

Reply:
That makes a lot of sense, I assume I can just add functions to the classes?

Reply:
No 🙃 from logging it looks like in some contexts we just have plain json.

![Screenshot 2024-05-19 at 22 29 28](https://github.com/maffin-io/maffin-app/assets/4065840/6a952b45-f62f-481d-8f0e-902e291b4ad4)

![Screenshot 2024-05-19 at 22 25 29](https://github.com/maffin-io/maffin-app/assets/4065840/cc2cb538-de70-4392-ab22-9ded8b3950ca)

Also the Split type either has an error or exposes an error. It says a split always has an account, but you (very defensively) write: `splits[0].account?.commodity?.guid === splits[1].account?.commodity?.guid`
A split clearly doesn't always have an account because 8 tests fail with "TypeError: Cannot read properties of undefined (reading 'commodity')" I make that assumption. So is the problem in the type or the runtime?

Reply:
> No 🙃 when we get form values like form.getValues('splits.0.fk_account') as Account we get plain json instead of the class instance.

Ah... I know why: https://github.com/maffin-io/maffin-app/blob/master/src/components/forms/transaction/TransactionForm.tsx#L48

so... typically they are instances but not in the case of the splits. I'm a bit confused though coz with the `...split` the instance methods should also be forwarded? 🤔 

Anyway, I'm happy with 1) or 2)

> It says a split always has an account, but you (very defensively) write: splits[0].account?.commodity?.guid === splits[1].account?.commodity?.guid

The problem is that when creating a new transaction, you get partial state. As a developer we can pass `Partial<Split>` because we want to define some default values for the users automatically or also, just pass an empty split and then let the user fill everything. This is why the `splits[0].account?.commodity?.guid === splits[1].account?.commodity?.guid` is so defensive. At the end state, for sure, splits always MUST have an account but at the beginning, the user may not have selected an account yet. 

Not sure if I explained that well, happy to chat more about it and if there are better options :)

Reply:
> I'm a bit confused though coz with the ...split the instance methods should also be forwarded? 🤔

Not if that method comes from the prototype chain, `...` only copies properties without accessing `__prototype__` eg. "own properties". I don't know how TypeOrm works so I don't know where the boundaries are.

I suggest we make 1) a type `SplitJSON` that only contains the data properties, and 2) a function that is responsible for copying data from the instance to a new object because I doubt `...` is going to be enough. By having a function we have one place to change it if we find other issues.

> Anyway, I'm happy with 1) or 2)

Then I think we should do 2) because that allows us to group the functions that works on data in a without side effects and uses other entities' operations: `Split.operations.isSameCommodity(split1, split2)`
This can be very useful when debugging because "operations" keyword will tell you that this is pure, so you can rule out database interactions and direct instance mutations.

Reply:
> I suggest we make 1) a type SplitJSON that only contains the data properties, and 2) a function that is responsible for copying data from the instance to a new object because I doubt ... is going to be enough. By having a function we have one place to change it if we find other issues.

Ok, let's go with the json type for Split and see how it works.

> Then I think we should do 2) because

maybe even the function to transform an instance to an object could be part of the operations too? But yeah sounds good to me :)